### PR TITLE
Input rewrite for touch screen

### PR DESCRIPTION
**Pull request by guusw**
- Detaches UI render/input processing
- Makes Inputs.Detached run on input thread and output last results on main thread

### CODE REVIEW

File: shards/core/brancher.hpp [hunk 1]
```diff
@@ -41,38 +45,43 @@ struct Brancher {
...
+        if (ignored.find(avail.name) != ignored.end())
+          continue;
         SHLOG_TRACE("Branch: adding variable to requirements: {}", avail.name);
         _mergedRequirements.push_back(avail);
       }
     } else {
       for (auto &avail : data.shared) {
         auto it = _collectedRequirements.find(avail.name);
         if (it != _collectedRequirements.end()) {
+          if (ignored.find(avail.name) != ignored.end())
```
Comment (sinkingsugar) [inline shards/core/brancher.hpp:75]: What's this ignored?

Comment (guusw) [reply shards/core/brancher.hpp]: Values that shouldn't be exposed inside the branch, although I think I don't need it anymore as I'm filtering object variables out of the capturing brancher now

Comment (sinkingsugar) [reply shards/core/brancher.hpp]: yeah adds complexity

File: shards/core/brancher.hpp [hunk 2]
```diff
@@ -40,19 +49,52 @@ struct Brancher {
   SHExposedTypesInfo requiredVariables() { return (SHExposedTypesInfo)_mergedRequirements; }
 
   const shards::ExposedInfo &getMergedRequirements() const { return _mergedRequirements; }
+  const std::unordered_set<std::string_view> &getCopyBySerialize() const { return _copyBySerialize; }
+
+  struct ShareableObjectFlags {
+    bool sharable{};
+    bool copyBySerialize{};
+  };
+  ShareableObjectFlags getSharableObjectFlags(const SHTypeInfo &type, bool shareObjectVariables) const {
```
Comment (sinkingsugar) [inline shards/core/brancher.hpp:58]: Can you add more comments to this file, why certain things are done etc

File: shards/core/capturing_brancher.hpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#include "brancher.hpp"
+#include "core/foundation.hpp"
+#include "shards/shards.h"
+#include <shards/modules/core/serialization.hpp>
+#include <string_view>
+#include <unordered_map>
+
+namespace shards {
+
+// Variant of brancher that captures variables
```
Comment (sinkingsugar) [inline shards/core/capturing_brancher.hpp:13]: You mean capture by deep copy right?

File: shards/egui/src/egui_host.rs [hunk 4]
```diff
@@ -153,23 +150,28 @@ impl EguiHost {
           return Err(e);
         }
 
-        self.full_output = Some(make_native_full_output(
-          &ui_ctx.egui_ctx,
-          egui_output,
-          draw_scale,
-        )?);
+        self.full_output = Some(egui_output);
+        // self.full_output = Some(make_native_full_output(
```
Comment (sinkingsugar) [inline shards/egui/src/egui_host.rs:154]: Delete? Or?

**Review by sinkingsugar — COMMENTED** [review]
Lots of changes! Nice work. I saw you deleted some tests, such as anims one, is that fine? Do we cover?

**Pull request merged**